modules/openfast-library/src/FAST_Subs.f90

```diff
@@ -5561,11 +8017,77 @@ SUBROUTINE FAST_Solution(t_initial, n_t_global, p_FAST, y_FAST, m_FAST, ED, BD,
+
+   TYPE(ElastoDyn_Data),     INTENT(IN   ) :: ED                  !< ElastoDyn data
+   TYPE(BeamDyn_Data),       INTENT(IN   ) :: BD                  !< BeamDyn data
+   TYPE(ServoDyn_Data),      INTENT(IN   ) :: SrvD                !< ServoDyn data
+   TYPE(AeroDyn14_Data),     INTENT(IN   ) :: AD14                !< AeroDyn14 data
+   TYPE(AeroDyn_Data),       INTENT(IN   ) :: AD                  !< AeroDyn data
+   TYPE(ExtLoads_Data),      INTENT(IN   ) :: ExtLd               !< External loads data
+   TYPE(InflowWind_Data),    INTENT(IN   ) :: IfW                 !< InflowWind data
+   TYPE(ExternalInflow_Data),INTENT(IN   ) :: ExtInfw             !< ExternalInflow data
+   TYPE(SCDataEx_Data),      INTENT(IN   ) :: SC_DX               !< Supercontroller Exchange data
+   TYPE(SeaState_Data),      INTENT(IN   ) :: SeaSt               !< SeaState data
+   TYPE(HydroDyn_Data),      INTENT(IN   ) :: HD                  !< HydroDyn data
+   TYPE(SubDyn_Data),        INTENT(IN   ) :: SD                  !< SubDyn data
+   TYPE(ExtPtfm_Data),       INTENT(IN   ) :: ExtPtfm             !< ExtPtfm_MCKF data
+   TYPE(MAP_Data),           INTENT(IN   ) :: MAPp                !< MAP data
+   TYPE(FEAMooring_Data),    INTENT(IN   ) :: FEAM                !< FEAMooring data
+   TYPE(MoorDyn_Data),       INTENT(IN   ) :: MD                  !< Data for the MoorDyn module
+   TYPE(OrcaFlex_Data),      INTENT(IN   ) :: Orca                !< OrcaFlex interface data
+   TYPE(IceFloe_Data),       INTENT(IN   ) :: IceF                !< IceFloe data
+   TYPE(IceDyn_Data),        INTENT(IN   ) :: IceD                !< All the IceDyn data used in time-step loop
+
+   TYPE(FAST_ModuleMapType), INTENT(IN   ) :: MeshMapData         !< Data for mapping between modules
+
+   INTEGER(IntKi),           INTENT(  OUT) :: ErrStat             !< Error status of the operation
+   CHARACTER(*),             INTENT(  OUT) :: ErrMsg              !< Error message if ErrStat /= ErrID_None
+
+   ! local variables
+   INTEGER(IntKi)                          :: I, k                ! generic loop counters
+
+   INTEGER(IntKi)                          :: ErrStat2
+   CHARACTER(ErrMsgLen)                    :: ErrMsg2
+   CHARACTER(*), PARAMETER                 :: RoutineName = 'FAST_WriteOutput'
+
+
+   ErrStat = ErrID_None
+   ErrMsg  = ""
+
 
    !----------------------------------------------------------------------------------------
    !! Check to see if we should output data this time step:
-   !----------------------------------------------------------------------------------------   
-   CALL WriteOutputToFile(n_t_global_next, t_global_next, p_FAST, y_FAST, ED, BD, AD14, AD, IfW, ExtInfw, SeaSt, HD, SD, ExtPtfm, &
+   !----------------------------------------------------------------------------------------
+   CALL WriteOutputToFile(n_t_global, m_FAST%t_global, p_FAST, y_FAST, ED, BD, AD14, AD, IfW, ExtInfw, SeaSt, HD, SD, ExtPtfm, &
```

Comment:
Not sure why you are passing variable `t_global` when using `m_FAST%t_global` in this routine. The call from `FAST_WriteOutput_T` uses `t_initial`, so that is also confusing.

Reply:
There was some confusion on this during the merge.  I'll get this fixed.